### PR TITLE
feat: Add support for Checkout API, new endpoints, more parameters

### PR DESCRIPTION
**Pull request by JiiHu**
I noticed that the SDK is missing support for Checkout API endpoints, so I added the basic support for that. Now you can use the SDK to fetch also details of Checkout orders. [Paddle Documentation of the added endpoint](https://developer.paddle.com/api-reference/fea392d1e2f4f-get-order-details)

I also added few other endpoints, refactored the code and added some missing params to existing endpoints

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "paddle-sdk",
   "description": "The Paddle.com Node.js SDK",
-  "version": "2.5.1",
+  "version": "2.5.3",
```
Comment (avaly) [inline package.json:4]: Please revert this line change.

File: sdk.js [hunk 2]
```diff
@@ -23,12 +26,11 @@ class PaddleSDK {
 	 * const client = new PaddleSDK('your-vendor-id', 'your-unique-api-key', 'your-public-key');
 	 */
 	constructor(vendorID, apiKey, publicKey, options) {
+		this.forCheckout = options && options.checkout;
```
Comment (avaly) [inline sdk.js:29]: This does not seem to be used.

File: sdk.js [hunk 3]
```diff
@@ -42,19 +44,33 @@ class PaddleSDK {
...
-	_request(path, { body = {}, headers = {}, form = true, json = false } = {}) {
-		const url = this.server + path;
+	_request(
+		path,
+		{
+			body = {},
+			headers = {},
+			form = true,
+			json = false,
+			takeFirst = false,
```
Comment (avaly) [inline sdk.js:54]:
```suggestion
			returnFirstItem = false,
```

File: sdk.js [hunk 4]
```diff
@@ -42,19 +44,33 @@ class PaddleSDK {
...
+			isCheckoutRequest = false,
```
Comment (avaly) [inline sdk.js:55]:
```suggestion
			checkoutAPI = false,
```

File: sdk.js [hunk 5]
```diff
@@ -82,6 +102,22 @@ class PaddleSDK {
 		};
 	}
 
+	/**
+	 * Get the used server URL. Some of the requests go to Checkout server, while most will go to Vendor server.
+	 */
+	_getServer(isCheckoutRequest) {
```
Comment (avaly) [inline sdk.js:108]:
```suggestion
	_serverURL(checkoutAPI = false) {
```

File: sdk.js [hunk 6]
```diff
@@ -146,6 +182,24 @@ class PaddleSDK {
 		});
 	}
 
+	/**
+	 * Get the plan based on its ID
+	 *
+	 * @method
+	 * @param {number} [planId]
+	 * @returns {Promise}
+	 * @fulfil {object} - The plans list
```
Comment (avaly) [inline sdk.js:191]: This comment is incorrect since we're returning only one plan.

File: sdk.js [hunk 7]
```diff
@@ -319,6 +373,26 @@ class PaddleSDK {
 		}
 	}
 
+	/**
+	 * Get subscription details
+	 *
+	 * @method
+	 * @param {number} subscriptionID
+	 * @returns {Promise}
+	 * @fulfill {object} - The result of the operation
```
Comment (avaly) [inline sdk.js:382]: This comment also needs to be updated.

**Review by avaly — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Just a few changes before we can merge this.

File: __tests__/attributes.test.js [hunk 8]
```diff
@@ -7,7 +7,7 @@ describe('attributes', () => {
 		expect(instance.vendorID).toBe('foo');
 		expect(instance.apiKey).toBe('bar');
 		expect(instance.publicKey).toBe('MISSING');
-		expect(instance.server).toBe('https://vendors.paddle.com/api/2.0');
+		expect(instance._getServer()).toBe('https://vendors.paddle.com/api/2.0');
```
Comment (avaly) [inline __tests__/attributes.test.js:10]: These will need to be updated as well.

File: sdk.js [hunk 9]
```diff
@@ -394,6 +484,39 @@ class PaddleSDK {
 		});
 	}
 
+	/**
+	 * Get the list of all users
+	 *
+	 * @method
+	 * @param {number} [planID]
```
Comment (avaly) [inline sdk.js:491]: Please update this parameter to reflect the implementation code (follow the format from the `constructor` jsdoc).

**Review by avaly — CHANGES_REQUESTED** [review]
Please remove `package-lock.json`.

Please add new unit tests for the new public methods.

File: __tests__/attributes.test.js (hunk 8 above)
Comment (JiiHu) [reply __tests__/attributes.test.js:10]: Thanks for these comments! I'll try to resolve them soon'ish.

**Comment by JiiHu** [comment]
@avaly This should be updated now! Is there still something to improve? :)

File: sdk.js [hunk 10]
```diff
@@ -355,7 +428,15 @@ class PaddleSDK {
...
-		const { quantity, price, planID, currency } = postData;
+		const {
+			quantity,
+			price,
+			planID,
+			currency,
+			prorate,
+			keepModifiers,
+			billImmediately,
+		} = postData;
```
Comment (avaly) [inline sdk.js:439]: Add the new attributes in the JSDoc above

**Review by avaly — CHANGES_REQUESTED** [review]
Please revert the `yarn.lock` changes in this PR.

That should be the last change required.

File: sdk.js (hunk 10 above)
Comment (JiiHu) [reply sdk.js:439]: I tried updating the `yarn.lock` as there were packages with security issues. Is there some way I could update the packages to secure ones in another way then?

Comment (avaly) [reply sdk.js:439]: I understand, but please create a separate PR for that update.

**Comment by JiiHu** [comment]
@avaly okay! How does this PR look now?

**Comment by avaly** [comment]
Just remove the `yarn.lock` changes and the PR will be ready to merge.

**Comment by rovaniemi** [comment]
Any updates @avaly?

**Comment by avaly** [comment]
@JiiHu do you have time to revert those `yarn.lock` changes?

**Comment by nadilas** [comment]
@JiiHu nice job! Can't wait to see this merged

**Comment by JiiHu** [comment]
@avaly Sorry, I missed this totally. How about now? 

@nadilas Thanks!

**Review by avaly — APPROVED** [review]

**Pull request merged**